include/ginkgo/core/preconditioner/jacobi.hpp

```diff
@@ -604,6 +608,7 @@ class Jacobi : public EnableLinOp<Jacobi<ValueType, IndexType>>,
     size_type num_blocks_;
     Array<value_type> blocks_;
     Array<remove_complex<value_type>> conditioning_;
+    std::shared_ptr<matrix::Diagonal<value_type>> diag_;
```

Comment:
How about reusing the `blocks` array instead? This avoids a few issues related to copy/move_to